cmd/gotk/create_source_git.go

```diff
@@ -237,8 +244,13 @@ func createSourceGitCmdRun(cmd *cobra.Command, args []string) error {
 	logger.Generatef("generating source")
 
 	if withAuth {
+		secretName := name
+		if sourceGitSecretRef != "" {
+			secretName = sourceGitSecretRef
+		}
```

Comment:
Instead of overwriting the generated secret here, I would expect the command to not generate a and upsert a secret at all if an existing secret name is given.

Reply:
So drop scenario 2 above?  Can do if that's what you prefer!

Reply:
It does not do scenario 2 at the moment, as on L234 the secret that is upserted does not take the configured value into account.

Reply:
🤦 Oops missed it, sorry.  Gimme a mo

Reply:
Adding just the name in there would not be sufficient by the way, as this would result in the overwrite of existing data.

It would require some additional logic to only create (and not update) a secret with the given name if it does not exist yet.

Reply:
What's the current behaviour if a secret with the same name as the git source exists?  Would that not have the same behavior?

Maybe I should just simplify this to just scenario 1?

Reply:
Let me know what you would prefer :)

Reply:
Please drop option 2. I see no reason why would you want to setup a name for the generated secret. When `secret-ref` is specified, then we assume it's up to the user to create it on the cluster by other means than gotk. If we make this change then it should be for all `create` commands not only Git.